### DISCUSSION TITLE
Seg fault parsing fixed length arrays in json_parser.c

### DISCUSSION THREAD

**Discussion by jolivem** [dvidelabs/flatcc#257] [General]
Hi,
I got a segmentation fault, file json_parser.c 
   line 882:        memset(s, 0, n - k);
Indeed, k has already been substracted from n few lines before, so:
   line 882:        memset(s, 0, n);
seems better for me.

**Comment by mikkelfj** [comment]
Indeed, just curious, this seems to be an error condition, so what input did you have to trigger this problem - in summary?

**Comment by mikkelfj** [comment]
I'm guessing you were parsing into a fixed length string in a struct, and the string was too short?

**Comment by mikkelfj** [comment]
Fix in 27a63fb on master.
